sdk/storage/Azure.Storage.Files.DataLake/src/PathClient.cs

```diff
@@ -0,0 +1,1462 @@
+                    nameof(BlobBaseClient),
+                    message:
+                    $"{nameof(Uri)}: {Uri}\n" +
+                    $"{nameof(recursive)}: {recursive}\n" +
+                    $"{nameof(conditions)}: {conditions}");
+                try
+                {
+                    Response<PathDeleteResult> response = await DataLakeRestClient.Path.DeleteAsync(
+                        clientDiagnostics: _clientDiagnostics,
+                        pipeline: Pipeline,
+                        resourceUri: _dfsUri,
+                        recursive: recursive,
+                        leaseId: conditions?.LeaseId,
+                        ifMatch: conditions?.IfMatch,
+                        ifNoneMatch: conditions?.IfNoneMatch,
+                        ifModifiedSince: conditions?.IfModifiedSince,
+                        ifUnmodifiedSince: conditions?.IfUnmodifiedSince,
+                        async: async,
+                        cancellationToken: cancellationToken)
+                        .ConfigureAwait(false);
+
+                    return response.GetRawResponse();
+                }
+                catch (Exception ex)
+                {
+                    Pipeline.LogException(ex);
+                    throw;
+                }
+                finally
+                {
+                    Pipeline.LogMethodExit(nameof(PathClient));
+                }
+            }
+        }
+        #endregion Delete
+
+        #region Rename
+        /// <summary>
+        /// The <see cref="Rename"/> operation renames a file or directory.
+        ///
+        /// For more information, see https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/create.
+        /// </summary>
+        /// <param name="destinationPath">
+        /// The destination path to rename the path to.
+        /// </param>
+        /// <param name="destConditions">
+        /// Optional <see cref="DataLakeRequestConditions"/> to add
+        /// conditions on the creation of this file or directory.
+        /// </param>
+        /// <param name="sourceConditions">
+        /// Optional <see cref="DataLakeRequestConditions"/> to add
+        /// conditions on the source on the creation of this file or directory.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Response{PathInfo}"/> describing the
+        /// newly created page blob.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="RequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        [ForwardsClientCalls]
+        public virtual Response<PathClient> Rename(
+            string destinationPath,
+            DataLakeRequestConditions destConditions = default,
```

Comment:
It should be `destinationConditions` and why does it come before `sourceConditions`?

Reply:
Will change.

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/8382